### PR TITLE
Workaround on moving vw_qgep_wastewater_structure

### PR DESCRIPTION
**Pull request by signedav**
It's done by converting the point to 2d, move it and add the Z value again.

This workaround has to be done, because if Z is 'NaN`, `St_Translate` would return `nan nan nan`.
```
SELECT ST_AsText( ST_TRANSLATE(ST_SetSRID(ST_MakePoint(500, 900, 'NaN'), 2056), 10, 20, 0 ) )
```
leads to:
```
"POINT Z (nan nan nan)"
```

### CODE REVIEW

File: delta/delta_1.2.1_fix_vw_qgep_wastewater_structure.sql [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+    UPDATE qgep_od.reach RE
+    SET progression_geometry =
+      ST_ForceCurve( ST_SetPoint(
+        ST_CurveToLine( RE.progression_geometry ),
+        ST_NumPoints(RE.progression_geometry) - 1,
+        ST_SetSRID( ST_MakePoint(
+            ST_X(ST_TRANSLATE(ST_MakePoint(ST_X(ST_EndPoint(RE.progression_geometry)), ST_Y(ST_EndPoint(RE.progression_geometry))), dx, dy )),
+            ST_Y(ST_TRANSLATE(ST_MakePoint(ST_X(ST_EndPoint(RE.progression_geometry)), ST_Y(ST_EndPoint(RE.progression_geometry))), dx, dy )),
+            ST_Z(ST_PointN(RE.progression_geometry, 1))), 2056 )
```
Comment (ponceta) [inline delta/delta_1.2.1_fix_vw_qgep_wastewater_structure.sql]: 2056 -> %(SRID)s ?

File: view/vw_qgep_wastewater_structure.sql [hunk 2]
```diff
@@ -554,7 +568,10 @@ BEGIN
       ST_ForceCurve( ST_SetPoint(
         ST_CurveToLine( RE.progression_geometry ),
         ST_NumPoints(RE.progression_geometry) - 1,
-        ST_TRANSLATE(ST_EndPoint(RE.progression_geometry), dx, dy )
+        ST_SetSRID( ST_MakePoint(
+            ST_X(ST_TRANSLATE(ST_MakePoint(ST_X(ST_EndPoint(RE.progression_geometry)), ST_Y(ST_EndPoint(RE.progression_geometry))), dx, dy )),
+            ST_Y(ST_TRANSLATE(ST_MakePoint(ST_X(ST_EndPoint(RE.progression_geometry)), ST_Y(ST_EndPoint(RE.progression_geometry))), dx, dy )),
+            ST_Z(ST_PointN(RE.progression_geometry, 1))), 2056 )
```
Comment (ponceta) [inline view/vw_qgep_wastewater_structure.sql]: 2056 -> %1$s ?

**Comment by ponceta** [comment]
Fix https://github.com/QGEP/QGEP/issues/476 

Thanks!

**Comment by ponceta** [comment]
@lbartoletti could you have a short look here and give it a go?

**Comment by lbartoletti** [comment]
@signedav @ponceta LGTM. Thanks

**Comment by m-kuhn** [comment]
Thanks @signedav @ponceta @lbartoletti

**Pull request merged**